playbooks/PnP_Workflow_Manager_Playbook.yml

```diff
@@ -0,0 +1,111 @@
+---
+- name: Manage operations - Add, claim, and delete devices of Onboarding Configuration (PnP)
+  hosts: localhost
+  connection: local
+  gather_facts: no
+
+  vars_files:
+    - "{{ CLUSTERFILE }}"
+
+  vars:
+    dnac_login: &dnac_login
+      dnac_host: "{{ dnac_host }}"
+      dnac_username: "{{ dnac_username }}"
+      dnac_password: "{{ dnac_password }}"
+      dnac_verify: "{{ dnac_verify }}"
+      dnac_port: "{{ dnac_port }}"
+      dnac_version: "{{ dnac_version }}"
+      dnac_debug: "{{ dnac_debug }}"
+      dnac_log_level: DEBUG
+
+  tasks:
+
+    - name: Import devices in bulk
+      cisco.dnac.pnp_workflow_manager:
+        <<: *dnac_login
+        dnac_log: True
+        state: merged
+        config_verify: True
+        config:
+            - device_info:
+              - serial_number: QD2425L8M7
+                state: Unclaimed
+                pid: c9300-24P
+                is_sudi_required: False
+
+              - serial_number: QTC2320E0H9
+                state: Unclaimed
+                pid: c9300-24P
+                hostname: Test-123
+
+              - serial_number: ETC2320E0HB
+                state: Unclaimed
+                pid: c9300-24P
+
+    - name: Add a new device and claim it
+      cisco.dnac.pnp_workflow_manager:
+        <<: *dnac_login
+        dnac_log: True
+        state: merged
+        config:
+          - site_name: Global/USA/San Francisco/BGL_18
+            device_info:
+              - serial_number: FJC2330E0BB
+                hostname: Test-9300-10
+                state: Unclaimed
+                pid: c9300-24P
+                is_sudi_required: True
+
+    - name: Claim an added Switch with template and image upgrade to a site only
+      cisco.dnac.pnp_workflow_manager:
+        <<: *dnac_login
+        dnac_log: True
+        state: merged
+        config:
+          - site_name: Global/USA/San Francisco/BGL_18
+            template_name: "Ansible_PNP_Switch"
+            image_name: cat9k_iosxe_npe.17.03.07.SPA.bin
+            project_name: Onboarding Configuration
+            template_details:
+              hostname: SJC-Switch-1
+              interface: TwoGigabitEthernet1/0/2
+            device_info:
+              - serial_number: FJC271924EQ
+                hostname: Switch
+                state: Unclaimed
+                pid: C9300-48UXM
+
+    - name: Claim an added Wireless Controller with template and image upgrade to a site only
```

Comment:
- name: Claim an existing Wireless Controller, apply a template, and upgrade its image for a specified site

Reply:
Done